### ISSUE TITLE
False Positive: UnusedLocalVariableCheck

### ISSUE THREAD

**Issue by Vyom-Yadav** [checkstyle/checkstyle#11374]
I have read check documentation: https://checkstyle.sourceforge.io/config_coding.html#UnusedLocalVariable
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac Test.java

/var/tmp $ cat config.xml
```
```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="UnusedLocalVariable"/>
    </module>
</module>
```
```
/var/tmp $ cat Test.java
```
```java
public class Test {

    static class m {
        static class h {
            public int a = 12;
        }
    }

    class j {
        void method() {
            int a = 22;
            m.h obj = new m.h() {
                int j = a + 12;
            };
            obj.getClass();
        }

        static class m {
            static class h {
            }
        }
    }
}
```
```
/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.0-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /var/tmp/Test.java:11:13: Unused local variable 'a'. [UnusedLocalVariable]
Audit done.
Checkstyle ends with 1 errors.

```
---

**Describe what you expect in detail.**
```
/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.0-all.jar -c config.xml Test.java
Starting audit...
Audit done.
```

---

Discovered while working on #11365, the source has been identified, the fix has been done locally, will be pushed soon.

**Comment by Vyom-Yadav** [comment]
Approved? This may be a non-common case but the check has support for these types of cases.

**Comment by nick-mancuso** [comment]
I agree that this is a false positive, please add proof of unused variable to issue description (i.e. print value of `j`) to make it obvious to maintainers.

**Comment by Vyom-Yadav** [comment]
Done.

**Comment by Vishakhakumawat** [comment]
I am on it.

**Comment by Vyom-Yadav** [comment]
@Vishakhakumawat Fix has already been done, just pushing to remote is remaining, it is mentiones in issue description

**Comment by Vishakhakumawat** [comment]
Hey can you help me actually I am new to open source and I have just
started so I don't have much idea about the checkstlye project actually
it's very large and I don't know to into which file I need to look in to
find issues .

**Comment by Vyom-Yadav** [comment]
https://github.com/checkstyle/checkstyle/wiki/To-GSoC-participant
